src/TropicalGeometry/matrix.jl

```diff
@@ -41,3 +41,37 @@ end
 function det(A::Matrix{TropicalSemiringElem{minOrMax}}) where {minOrMax<:Union{typeof(min),typeof(max)}}
     return det(matrix(TropicalSemiring{minOrMax},A))
 end
+
+@doc raw"""
+    is_tropically_generic(A::MatrixElem{<:TropicalSemiringElem}, minOrMax)
+
+Check if a collection of vectors in the tropical torus (given as rows of a matrix `A`) are in tropical general position with respect to the `minOrMax` convention.
+
+# Examples
+```jldoctest
+julia> A = matrix(tropical_semiring(),[1 0;0 1])
+[(1)   (0)]
+[(0)   (1)]
+
+julia> is_tropically_generic(A,min)
+true
+```
+"""
+function is_tropically_generic(A::MatrixElem{<:TropicalSemiringElem}, minOrMax)
+  @req convention(A) == minOrMax "Semiring convention not as declared"
```

Comment:
Naive question (sorry if it is obvious for domain experts): given this required equality, why is there a `minOrMax` argument at all?

Reply:
I think the `minOrMax` argument is indeed not necessary and should be removed.

Reply:
It may be true that it is not necessary, but I will at least explain why we put it in. Points are in tropical generic position iff the tropical hyperplane arrangement (where each point is the apex of a hyperplane) is in generic position. Now someone may describe the points using MIN tropical numbers and then ask if they describe a generic MAX hyperplane arrangement. In that case it becomes confusing since tsgn or tdet of a matrix described by MIN tropical numbers will do the computation in MIN. So we put this argument in to make sure people know which hyperplane arrangement they are getting the answer for. Sorry if this is not so clear. I will describe the example that made us do this.
$P = new Polytope<Min>(POINTS=>[[0,1,-1,3],[0,1,-3,3],[0,1,-1,0],[0,-1,-1,3]]);
P is a polytrope where the vertices are not in tropically generic position.
Looking at its points in a matrix
$V = new Matrix<TropicalNumber<Min>>([[0,0,0,0],[1,1,1,-1],[-1,-3,-1,-1],[3,3,0,3]]);
I got tsgn($V) = 1
This confused us initially since the points are definitely not in generic position from the point of view of the tropical convex hull. But the MIN tropical convex hull actually uses MAX tropical hyperplanes (or lets say is the same as them). When I redefined V as
$V = new Matrix<TropicalNumber<Max>>([[0,0,0,0],[1,1,1,-1],[-1,-3,-1,-1],[3,3,0,3]]);
I got the expected tsgn($V) = 0
So what we figured was we would tell the user to double check whether they want a MAX or a MIN hyperplane arrangement. Then they make sure that the numbers in their matrix are the same as which hyperplane arrangement they want, so that tsgn can do what the user wants it to do.

Reply:
Then how about creating two versions of `is_tropically_generic`:

(1) the input is a matrix over the tropical semiring (no second argument)
(2) the input is a matrix over the rationals and either min or max

Function (2) then converts the rational matrix to a tropical semiring matrix and calls Function (1).

Reply:
That way, if you have a rational matrix like the vertices of a polytope, you either have to 
(a) explicitly cast it into a tropical semiring (and thereby choosing a convention) or 
(b) explicitly state the convention when calling `is_tropically_generic`.

This should help people avoid the confusion you had in polymake.

Reply:
I think that's not a bad idea. The only sticking point is the quirk in tropical convexity that the MIN tropical convex hull is the same as the MAX tropical hyperplane arrangement. So if someone creates a MIN tropical number matrix for the points of a convex hull computation, they then have to make a new matrix with MAX tropical numbers, so that they can see if the points are generic. And even though they did declare MIN when making the matrix for the points, they may miss that they have to make a new MAX matrix...in which case the "is_tropically_generic" function may tell them yes your points are in generic position, even though for their purposes it is not. When I made this mistake the only reason I knew there was something going wrong was that the visualization made it clear that the points are not in tropically generic position.

I think you are onto something with the two versions of is_tropically_generic, but now I have the idea of maybe making the two functions `is_max_tropically_generic` and `is_min_tropically_generic`. We can throw an error if someone plugs in MIN numbers into `is_max_tropically_generic` and visa versa. 

Anyway, it would be perfectly fine to go with your suggestion. I am very new to this, so I trust your judgement.

Reply:
One question:  Why is it important for a min-tropical convex hull that the max-tropical hyperplane arrangement is generic?

Reply:
One answer is, suppose I want to make a polytrope (under min tropical convex hull) that has the maximal number of vertices. To do this the points must have a generic max-tropical hyperplane arrangement. Another would be, that I want to test a hypothesis on non-generic polytropes, and I want a way to see if they are generic or not.

The main point is that in this context min and max have to be used interchangeably. 
A point configuration in the tropical torus induces a polyhedral subdivision of the tropical torus. The bounded cells of this subdivision are the min tropical convex hull of that point configuration. At the same time that polyhedral subdivision arises from the max-tropical hyperplane arrangement of the point configuration. One way to see if the subdivision is generic (and that the min tropical convex hull is generic) is to see if the max-tropical hyperplane arrangement is generic.

Reply:
> One answer is, suppose I want to make a polytrope (under min tropical convex hull) that has the maximal number of vertices. To do this the points must have a generic max-tropical hyperplane arrangement. Another would be, that I want to test a hypothesis on non-generic polytropes, and I want a way to see if they are generic or not.

I see, in that case user confusion can be avoided if we had a type `Polytrope` and functions 
* `has_maximal_number_of_vertices(::Polytrope)`
* `is_generic(::Polytrope)`

The problem is that matrices are too universal.  They represent a whole slew of mathematical objects and what may be natural for one interpretation might be counter-intuitive for the other.  The solution in OSCAR is just having different types for different mathematical objects, even if they can all represented by matrices.

So I'd still prefer my suggestion, but in the long term - if there is some demand - we should think of creating a type `Polytrope` together with basic functions thereon.

Reply:
> Then how about creating two versions of `is_tropically_generic`:
> 
> (1) the input is a matrix over the tropical semiring (no second argument) (2) the input is a matrix over the rationals and either min or max
> 
> Function (2) then converts the rational matrix to a tropical semiring matrix and calls Function (1).

Okay then implementing it like this sounds good. Do we still implement it though? Regarding your comment below about possibly not.